gl3n/aabb.d

```diff
@@ -102,62 +103,66 @@ struct AABBT(type) {
+        assert(AABB(vec3(cast(at)0.0, cast(at)0.0, cast(at)0.0), vec3(cast(at)1.0, cast(at)1.0, cast(at)1.0)).intersects(
+               AABB(vec3(cast(at)0.5, cast(at)0.5, cast(at)0.5), vec3(cast(at)1.7, cast(at)1.7, cast(at)1.7))));
 
-        assert(!AABB(vec3(0.0f, 0.0f, 0.0f), vec3(1.0f, 1.0f, 1.0f)).intersects(
-                AABB(vec3(1.5f, 1.5f, 1.5f), vec3(3.0f, 3.0f, 3.0f))));
+        assert(!AABB(vec3(cast(at)0.0, cast(at)0.0, cast(at)0.0), vec3(cast(at)1.0, cast(at)1.0, cast(at)1.0)).intersects(
+                AABB(vec3(cast(at)2.5, cast(at)2.5, cast(at)2.5), vec3(cast(at)3.0, cast(at)3.0, cast(at)3.0))));
     }
 
     /// Returns the extent of the AABB (also sometimes called size).
-    @property vec3 extent() const {
+    @property  extent() const {
         return max - min;
     }
 
     /// Returns the half extent.
     @property vec3 half_extent() const {
-        return 0.5 * (max - min);
+        return (max - min) / 2;
     }
 
     unittest {
-        AABB a = AABB(vec3(0.0f, 0.0f, 0.0f), vec3(1.0f, 1.0f, 1.0f));
-        assert(a.extent == vec3(1.0f, 1.0f, 1.0f));
-        assert(a.half_extent == 0.5 * a.extent);
-
-        AABB b = AABB(vec3(0.2f, 0.2f, 0.2f), vec3(1.0f, 1.0f, 1.0f));
-        assert(b.extent == vec3(0.8f, 0.8f, 0.8f));
-        assert(b.half_extent == 0.5 * b.extent);
+        alias AABBT!at AABB;
+        AABBT!at a = AABBT!at(vec3(cast(at)0.0, cast(at)0.0, cast(at)0.0), vec3(cast(at)10.0, cast(at)10.0, cast(at)10.0));
+        assert(a.extent == vec3(cast(at)10.0, cast(at)10.0, cast(at)10.0));
+        assert(a.half_extent == a.extent / 2);
+
+        AABBT!at b = AABBT!at(vec3(cast(at)2.0, cast(at)2.0, cast(at)2.0), vec3(cast(at)10.0, cast(at)10.0, cast(at)10.0));
+        assert(b.extent == vec3(cast(at)8.0, cast(at)8.0, cast(at)8.0));
+        assert(b.half_extent == b.extent / 2);
         
     }
 
     /// Returns the area of the AABB.
     @property at area() const {
         vec3 e = extent;
-        return 2.0 * (e.x * e.y + e.x * e.z + e.y * e.z);
+        return cast(at)(cast(at)2.0 * cast(at)(e.x * e.y + e.x * e.z + e.y * e.z));
     }
 
     unittest {
-        AABB a = AABB(vec3(0.0f, 0.0f, 0.0f), vec3(1.0f, 1.0f, 1.0f));
+        alias AABBT!at AABB;
+        AABB a = AABB(vec3(cast(at)0.0, cast(at)0.0, cast(at)0.0), vec3(cast(at)1.0, cast(at)1.0, cast(at)1.0));
         assert(a.area == 6);
 
-        AABB b = AABB(vec3(0.2f, 0.2f, 0.2f), vec3(1.0f, 1.0f, 1.0f));
-        assert(almost_equal(b.area, 3.84f));
+        AABB b = AABB(vec3(cast(at)2.0, cast(at)2.0, cast(at)2.0), vec3(cast(at)10.0, cast(at)10.0, cast(at)10.0));
+        assert(almost_equal(b.area, cast(at)384));
 
-        AABB c = AABB(vec3(0.2f, 0.4f, 0.6f), vec3(1.0f, 1.0f, 1.0f));
-        assert(almost_equal(c.area, 2.08f));
+        AABB c = AABB(vec3(cast(at)2.0, cast(at)4.0, cast(at)6.0), vec3(cast(at)10.0, cast(at)10.0, cast(at)10.0));
+        assert(almost_equal(c.area, cast(at)208.0));
```

Comment:
It turns out that the result of `cast(byte)208` is `127` in GDC 4.8.2.
Not sure how to proceed, please advice.